### PR TITLE
Db controllers

### PR DESCRIPTION
**Pull request by TannerTreasure**
Implemented getById, create, update and delete from skeleton in clubProfileController.js
- Assumes req.body from front end will be properly formatted for each function

### CODE REVIEW

File: server/models/ClubOverviewModel.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const mongoose = require('mongoose');
```
Comment (wangyyt1013) [inline server/models/ClubOverviewModel.js:1]: this PR and Sad's PR #63 are dead locking (waiting for each other); please consider removing changes related to that PR from this PR

File: server/controllers/clubProfileController.js [hunk 2]
```diff
@@ -6,15 +9,30 @@ module.exports = {
...
+
     },
     create: function(req, res) {
         // TODO; req.body contains profile information
+        clubProfile.create(req.body)
+                .then(newclubProfile => res.json(newclubProfile))
+                .catch(err => res.status(422).json(err))
     },
     update: function(req, res) {
         // TODO; req.body contains profile information and req.params.id
```
Comment (wangyyt1013) [inline server/controllers/clubProfileController.js:24]: have you confirmed the mongoose interface definition of `findOneAndUpdate` and `findById` and made sure that they are used this way? has any testing been performed?

File: server/controllers/clubProfileController.js [hunk 3]
```diff
@@ -6,15 +9,30 @@ module.exports = {
...
+        clubProfile.findOneAndUpdate({ _id: req.params.id}, req.body)
+                .then(clubprofile => res.json(clubprofile))
+                .catch(err => res.status(422).json(err))
     },
     delete: function(req, res) {
         // TODO: req.params.id
+        clubProfile.findById({ _id: req.params.id })
```
Comment (wangyyt1013) [inline server/controllers/clubProfileController.js:31]: what's the reason for using `findById` differently here vs. earlier in the `getById` function?

**Review by wangyyt1013 — CHANGES_REQUESTED** [review]
generally lgtm, although a few suggestions / clarifying questions!

File: server/models/ClubOverviewModel.js (hunk 1 above)
Comment (TannerTreasure) [reply server/models/ClubOverviewModel.js]: I've removed the overlapping changes

File: server/controllers/clubProfileController.js (hunk 2 above)
Comment (TannerTreasure) [reply server/controllers/clubProfileController.js:24]: I've confirmed the functionality of each function and made the usage consistent throughout.

File: server/controllers/clubProfileController.js (hunk 3 above)
Comment (TannerTreasure) [reply server/controllers/clubProfileController.js:31]: They are the same, get by Id would be used by the frontend. getById will have different usage in the future as discussed in out meeting

**Review by wangyyt1013 — APPROVED** [review]
This is awesome, thanks for the hard work!

**Pull request merged**